hermes/bin/src/runtime_extensions/hermes/cron/mod.rs

```diff
@@ -1,15 +1,582 @@
+) -> Vec<CronComponent> {
+    cron_time
+        .iter()
+        .map(|d| d.clamp_inner(min_val, max_val))
+        .collect()
+}
+
+/// Scan over the remaining components and merge them if they overlap
+fn merge_cron_time_overlaps(cron_time: Vec<CronComponent>) -> Vec<CronComponent> {
+    let mut cron_time = cron_time;
+    // Sort the clamped components to have a consistent order
+    cron_time.sort();
+
+    let merged = cron_time
+        .clone()
+        .iter()
+        .fold(Vec::new(), |mut out, cron_component| {
+            let has_no_overlap = out
+                .iter()
+                .all(|&item: &CronComponent| !item.overlaps(*cron_component));
+            if has_no_overlap {
+                out.push(*cron_component);
+            } else {
+                for item in &mut out {
+                    if item.overlaps(*cron_component) {
+                        if let Some(merged_item) = item.merge(*cron_component) {
+                            *item = merged_item;
+                        }
+                    }
+                }
+            }
+            out.sort();
+            out
+        });
+    merged
+}
+
+/// Convert a `CronTime` to a `CronSched` for the day of week.
+fn cron_time_to_cron_sched_dow(cron_time: &CronTime) -> CronSched {
+    cron_time_to_cron_sched(cron_time, CronComponent::MIN_DOW, CronComponent::MAX_DOW)
+}
+
+/// Convert a `CronTime` to a `CronSched` for the month.
+fn cron_time_to_cron_sched_month(cron_time: &CronTime) -> CronSched {
+    cron_time_to_cron_sched(
+        cron_time,
+        CronComponent::MIN_MONTH,
+        CronComponent::MAX_MONTH,
+    )
+}
+
+/// Convert a `CronTime` to a `CronSched` for the day of month.
+fn cron_time_to_cron_sched_day(cron_time: &CronTime) -> CronSched {
+    cron_time_to_cron_sched(cron_time, CronComponent::MIN_DAY, CronComponent::MAX_DAY)
+}
+
+/// Convert a `CronTime` to a `CronSched` for the hour of day.
+fn cron_time_to_cron_sched_hour(cron_time: &CronTime) -> CronSched {
+    cron_time_to_cron_sched(cron_time, CronComponent::MIN_HOUR, CronComponent::MAX_HOUR)
+}
+
+/// Convert a `CronTime` to a `CronSched` for the minute of hour.
+fn cron_time_to_cron_sched_minute(cron_time: &CronTime) -> CronSched {
+    cron_time_to_cron_sched(
+        cron_time,
+        CronComponent::MIN_MINUTE,
+        CronComponent::MAX_MINUTE,
+    )
+}
+
+impl CronComponent {
+    /// Represents all possible values, `"*"`.
+    const ALL_STR: &'static str = "*";
+    /// Maximum value for `Day`.
+    const MAX_DAY: u8 = 31;
+    /// Maximum value for `DayOfWeek`. Sunday.
+    const MAX_DOW: u8 = 7;
+    /// Maximum value for `Hour`.
+    const MAX_HOUR: u8 = 23;
+    /// Maximum value for `Minute`.
+    const MAX_MINUTE: u8 = 59;
+    /// Maximum value for `Month`.
+    const MAX_MONTH: u8 = 12;
+    /// Minimum value for `Day`.
+    const MIN_DAY: u8 = 1;
+    /// Minimum value for `DayOfWeek`. Monday.
+    const MIN_DOW: u8 = 0;
```

Comment:
Should this be 1 🤔

Reply:
Not here, at least following the same convention as https://www.man7.org/linux/man-pages/man5/crontab.5.html , the `DayOfWeek` range is defined as `0-7`.